Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -10337,8 +10337,16 @@ bool WebPageProxy::didChooseFilesForOpenPanelWithImageTranscoding(const Vector<S
 {
 #if PLATFORM(MAC)
     auto transcodingMIMEType = WebCore::MIMETypeRegistry::preferredImageMIMETypeForEncoding(allowedMIMETypes, { });
-    if (transcodingMIMEType.isNull())
-        return false;
+ß
+    RefPtr document = frame.document();
+    if (transcodingMIMEType.isNull()) {
+        // For designated sites which are sending "image/*", we need to force the Mimetype to be able to transcode HEIC.
+        // FIXME: How to pass document->quirks() here?
+        if (document->quirks().needsForcingImageTranscoding())
+            transcodingMIMEType = "image/jpeg"_s;
```

Comment:
This is not yet working. We need to have the context of the document or the URL to pass to the function… 
hmmm writing that I could probably just do that.

Reply:
We passed currentURL()